R/add_ppa_flag.R

```diff
@@ -0,0 +1,149 @@
+#' Add Potentially Preventable Admissions Marker
+#'
+#' @description This function takes a data frame input and determines, based on
+#' a combination of diagnostic codes and operation codes, whether an admission was
+#' preventable or not
+#' @param data A data frame
+#'
+#' @return A data frame to use as a lookup of PPAs
+#' @export
+add_ppa_flag <- function(data) {
+  matching_data <- data %>%
+    # Select out only the columns we need
+    dplyr::select(
+      .data$anon_chi, .data$cij_marker, .data$cij_pattype, .data$recid,
+      .data$op1a, .data$diag1, .data$diag2, .data$diag3, .data$diag4,
+      .data$diag5, .data$diag6
+    ) %>%
+    # Filter only recids and patient type where admission was preventable
+    dplyr::filter(.data$recid %in% c("01B", "02B", "04B", "GLS") & .data$cij_pattype == "Non-Elective") %>%
```

Comment:
Can you write some checks so we get a nice error message if any of the required variables are not submitted, please?

```r
if (!(all(vars %in% names(data)) {
   # Figure out which variables(s) were missing
   missing_vars <- ...

   cli::cli_abort("{missing_vars} were missing and are needed.")
}
```

Also, a check with a message/warning if none of those recids are provided (it will still work if we don't have those recids, it just won't flag any as PPA so it is a bit pointless).

Reply:
Good idea, done the suggested one and will add the other when I'm back

Reply:
Done, but will amend with use of #387 when merged